### PR TITLE
Tweak code flow on top of #8324

### PR DESCRIPTION
**Pull request by beorn7**
As promised in #8324.

### CODE REVIEW

File: cmd/promtool/backfill.go [hunk 1]
```diff
@@ -106,15 +115,22 @@ func createBlocks(input []byte, mint, maxt int64, maxSamplesInAppender int, outp
...
+					p.Metric(&l)
 					return errors.Errorf("expected timestamp for series %v, got none", l)
 				}
-				if *ts < t || *ts >= tsUpper {
+				if *ts < t {
 					continue
 				}
+				if *ts >= tsUpper {
+					nextSampleTs = *ts
+					break
```
Comment (roidelapluie) [inline cmd/promtool/backfill.go:129]: Do we have a guarantee that the samples are always ordered by timestamp? I had the impression that the ordering is only guaranteed per metric.

Comment (roidelapluie) [reply cmd/promtool/backfill.go:129]: If the ordering is guaranteed and we can break safely, then we would not need to read the complete file each time we want a block.

Comment (roidelapluie) [reply cmd/promtool/backfill.go:129]: We could just parse once.

Comment (beorn7) [reply cmd/promtool/backfill.go:129]: Of course. Damn you OpenMetrics! So what I saw as a needless complication was actually necessary. Will delete this branch quickly to hide my embarrassment.

**Pull request closed without merging**